src/Shared/UnitTests/EngineTestEnvironment.cs

```diff
@@ -20,7 +20,9 @@ namespace Microsoft.Build.UnitTests
     public partial class TestEnvironment
     {
         // reset the default build manager and the state it might have accumulated from other tests
+#pragma warning disable CA1823 // Avoid unused private fields
```

Comment:
I think you can just delete this? It looks legitimately unused.

Reply:
I thought that too, but I think the constructor calls to create a new singleton. Which then resets some things. Its very strangely structured.

Reply:
I tried deleting it and running tests, and it passed tests. Evidence that it isn't used but not definitive.

The constructor refers to SingletonField and by reflection to s_singletonInstance on BuildManager, but I didn't see a reference to _resetBuildManager.

Reply:
I actually have no idea. I felt it erred on the side of caution and used a suppression instead of deleting all the code.

So what I think is happening, is that the TestEnvironment class is used in different tests. When it is instantiated, it makes a call to the DefaultBuildManager to clean-up the environment.

This seems strange, because I would have thought this would have been done on dispose for the TestEnvironment since it is utilised in a using pattern.

The comment explains it best:
`// reset the default build manager and the state it might have accumulated from other tests`

Reply:
I get it. I guess the idea was that if the environment was corrupted (i.e., not used in a using pattern, there are bugs, etc.), someone might go down a rabbit hole trying to figure out why. Worse, it would only happen if the tests were run in a particular order. I might claim that this then hides potential serious problems, but I guess it's fine to leave it in.

Reply:
It should be safe to put

```C#
_ = new ResetDefaultBuildManager();
```

in the constructor and delete the field.

Reply:
@ladipro 
The constructor is here:
https://github.com/dotnet/msbuild/blob/597d77e17ea6b12e34c63f99daf3aafec76137e5/src/Shared/UnitTests/TestEnvironment.cs#L58-L63

But if I add it here, it cannot find the class when in the `Microsoft.Build.Framework.UnitTests` project. I am not sure how to get around this. Maybe a `#if` but if its used somewhere else it would have to be updated. So its not an elegant solution.

I don't suppose EngineTestEnvironment wasn't included in `Microsoft.Build.Framework.UnitTests` by accident?

Reply:
Ah, I see, let's leave as is then. Maybe the better way to structure the code would be to make the Engine `TestEnvironment` derive from the base `TestEnvironment` instead of using partial classes but definitely out of scope here. Apologies for the randomization.